oeps/best-practices/oep-0067-bp-tools-and-technology.rst

```diff
@@ -0,0 +1,342 @@
+OEP-67: Tools and Technology Standards
+######################################
+
+.. list-table::
+   :widths: 25 75
+
+   * - OEP
+     - Link to the doc in the following format::
+
+        :doc:`OEP-67 <oep-0067-bp-tools-and-technology>`
+
+   * - Title
+     - Tools and Technology Standards
+   * - Last Modified
+     - 2023-09-05
+   * - Authors
+     - Feanil Patel <feanil@axim.org>
+   * - Arbiter
+     - TBD
+   * - Status
+     - Accepted
+   * - Type
+     - Best Practice
+   * - Created
+     - 2023-09-05
+   * - Review Period
+     - 2023-09-05 - 2023-10-xx
+   * - Resolution
+     - <links to any discussions where the final status was decided>
+   * - References
+     - :doc:`oep-0011-bp-FED-technology`
+
+Abstract
+********
+
+The Open edX Platform is built using many 1st and 3rd party tools and technologies.
+This document covers the current recommendations for the tools and technologies
+that should be used.
+
+Motivation
+**********
+
+The Open edX platform has a lot of 1st and 3rd party tools and technology that it uses.
+The problems we're trying to solve are:
+
+    * It is difficult to know what the current best practice for a given area.
+
+    * It is unclear how to go about the discussion and decision making process
+      to change a best practice.
+
+:doc:`OEP-11 <oep-0011-bp-FED-technology>` serves as a great example for the
+usefulness of this kind of OEP but limits itself to be relevant only to
+frontend technologies.  This OEP would supersede :doc:`OEP-11
+<oep-0011-bp-FED-technology>` and encompass not only frontend technolog but any
+technology that:
+
+    * Is considered a best practice for the Open edX Platform
+
+    * Is relevant to a large subset of the Open edX Platform
+
+Specification
+*************
+
+.. note:: We'll fill this out before the OEP lands but leaving mostly blank for initial feedback.
+
+General Technology Selection
+============================
+
+This technology is not specific to frontend or backend code.
+
+#. **Codecov should be used to measure code covered by tests**
+
+   **Rationale**: It is important to measure the amount of code covered by our
+   automated test suites. By striving for a high level of test coverage, we can
+   reduce the number of bugs that can only be found via manual testing. another
+   added benefit comes when we include coverage runs in the CI process, as
+   contributors are automatically reminded to include tests for any new code.
+
+   The Open edX community has standardized around `Codecov`_ as a measurement tool.
```

Comment:
```suggestion
#. **Codecov should be used to report code coverage by unit tests**

   **Rationale**: It is important to measure the amount of code covered by our
   automated test suites. By striving for a high level of test coverage, we can
   reduce the number of bugs that can only be found via manual testing, 
   and by using codecov to run coverage in CI,
   contributors are automatically reminded to include tests for any new code.

   The Open edX community has standardized around `Codecov`_ as a coverage reporting service.
```

(Nitpicking this particular entry a bit, but I think this will be relevant for other technology choices too.)

Codecov feels different than, say, React, because it's a proprietary SaaS thing. If the offering starts to get bad, we will need to switch, even if we like the tools that come with the service. I think it's fine to list it alongside React et al, but I want to be clear that this isn't really a tool, it's a service.

And if Codecov does ever get bad, we can continue locally measuring Python code coverage using coverage.py (which might be what Codecov is using under the hood... not sure).

Reply:
I think it's useful to enumerate the specific services here because often many services can do the same job but picking one and listing it is a useful way to reduce fracturing across the ecosystem.  Also, if tomorrow Codecov got bad, we don't currently have a good place to have the conversation about what the alternatives are and why we would pick one over others.  I think this OEP will provide the place for that conversation for both tools and SaaS services we use as a community.